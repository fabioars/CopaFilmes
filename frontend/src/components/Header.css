.Header {
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    text-align: center;
    background-color: var(--bg-color-darken);
    color: #fff;
    min-height: 375px;
    margin-bottom: 40px;
}

.Header__title {
    position: relative;
    font-size: 38px;
    margin-top: 0;
    margin-bottom: 40px;
    order: 2;
}

.Header__title:after {
    content: '';
    position: absolute;
    left: 50%;
    bottom: -20px;
    display: block;
    width: 25px;
    height: 3px;
    background-color: var(--bg-color);
    transform: translateX(-50%);
}

.Header__subtitle {
    color: #A8A8A8;
    text-transform: uppercase;
    font-size: 18px;
    margin-bottom: 25px;
    order: 1;
}

.Header__description {
    max-width: 800px;
    order: 3
}
