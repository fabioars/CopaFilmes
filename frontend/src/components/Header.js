import React from 'react';
import './Header.css';

export default function Header({ title, subtitle, children }) {
    return (
        <header className="Header">
            <h1 className="Header__title">{title}</h1>
            <h2 className="Header__subtitle">{subtitle}</h2>

            <div className="Header__description">
                {children}
            </div>
        </header>
    );
}

Header.defaultProps = {
    title: '',
    subtitle: 'Campeonato de filmes',
    children: null,
};
