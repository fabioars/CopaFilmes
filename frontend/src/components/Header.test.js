import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Header from './Header';

describe('Header', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Header />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders correctly', () => {
        const tree = renderer.create(<Header/>).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with title', () => {
        const tree = renderer.create(<Header title="Fase de Seleção" />).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with children', () => {
        const tree = renderer.create(
            <Header title="Fase de Seleção">Uma descrição básica para o teste</Header>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
