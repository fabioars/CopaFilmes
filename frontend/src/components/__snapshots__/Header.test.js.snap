// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="Header"
>
  <h1
    className="Header__title"
  >
    
  </h1>
  <h2
    className="Header__subtitle"
  >
    Campeonato de filmes
  </h2>
  <div
    className="Header__description"
  />
</header>
`;

exports[`Header renders correctly with children 1`] = `
<header
  className="Header"
>
  <h1
    className="Header__title"
  >
    Fase de Seleção
  </h1>
  <h2
    className="Header__subtitle"
  >
    Campeonato de filmes
  </h2>
  <div
    className="Header__description"
  >
    Uma descrição básica para o teste
  </div>
</header>
`;

exports[`Header renders correctly with title 1`] = `
<header
  className="Header"
>
  <h1
    className="Header__title"
  >
    Fase de Seleção
  </h1>
  <h2
    className="Header__subtitle"
  >
    Campeonato de filmes
  </h2>
  <div
    className="Header__description"
  />
</header>
`;
